### ISSUE TITLE
"Duplicate strategy name" warning for worker superclass

### ISSUE THREAD

**Issue by mahemoff** [ixti/sidekiq-throttled#26]
Seeing a "Duplicate strategy name" error when putting throttle rules in a worker superclass.

> /rubies/ruby-2.3.1/lib/ruby/gems/2.3.02.3.0/gems/sidekiq-throttled-0.6.6/lib/sidekiq/throttled/registry.rb:23

I believe this happens when 2 or more worker classes are inheriting from a superclass with throttle. I'd like to keep things DRY with a single declaration of the throttle. In my case, I don't particularly care if the throttle applies separately to each subclass or it's "global" for everything in the hierarchy (as happens now I think, because they are keyed on the superclass name). Either way would be fine as long as it's well-defined and there's no warning.

**Comment by masterkain** [comment]
seeing this using normal classes

**Comment by cassiopagnoncelli** [comment]
What real world problem arises with this?

**Comment by jhrussell** [comment]
Not sure about runtime problems.  We have this warning, but runtime appears to operate correctly otherwise.  However, confusion and time wasted investigating and trying to address the warning are some of the problems I have encountered.

**Comment by cassiopagnoncelli** [comment]
I'm not sure whether it can have negative effects whilst running multiple Sidekiq processes

**Comment by kalemi19** [comment]
Have you added the following in your app's bootstrap?

```
require "sidekiq/throttled"
Sidekiq::Throttled.setup!
```

I was having the same issue, plus the throttler wasn't working at all, but once I added the above in `config/initializers/sidekiq.rb` everything started working properly.

**Comment by cassiopagnoncelli** [comment]
@kalemi19 yes, I have this Throttled.setup! for a long time, always get the same warnings

**Comment by SwannG** [comment]
> the throttle applies separately to each subclass or it's "global" for everything in the hierarchy

Does anyone know the answer to that?

**Comment by mahemoff** [comment]
Guessing it's global as it would probably be keyed on the class name. So if you had B and C as subclasses of A set to 1 per minute, only one of B or C could fire each minute. Maybe someone could confirm it works that way.

There's actually good arguments for it to work either way tbh, but personally I've just avoided putting it in superclasses.
